src/Http/Senders/SimulatedSender.php

```diff
@@ -70,6 +70,10 @@ public function sendRequest(PendingRequest $pendingRequest, bool $asynchronous =
 
         // We'll return the synchronous response directly
 
+        if($pendingRequest->getRequest()->delay()->all() > 0) {
```

Comment:
We should be checking the delay on the PendingRequest directly instead of getting the request through the PendingRequest `$pendingRequest->delay()->all()`

Same as my comment above, can we do `$pendingRequest->delay()->isNotEmpty()` ?

Reply:
Will add delay method on pending request.